source/include/ice_api.h

```diff
@@ -0,0 +1,105 @@
+#ifndef ICE_API_H
+#define ICE_API_H
+
+#include "ice_data_types.h"
+
+IceResult_t Ice_CreateIceAgent( IceAgent_t * pIceAgent,
+                                char * pLocalUsername,
+                                char * pLocalPassword,
+                                char * pRemoteUsername,
+                                char * pRemotePassword,
+                                char * pCombinedUsername,
+                                TransactionIdStore_t * pBuffer,
+                                Ice_ComputeRandom computeRandomFunction,
+                                Ice_ComputeCrc32 computeCrc32Function,
+                                Ice_ComputeHMAC computeHMACFunction );
+
+IceResult_t Ice_AddHostCandidate( const IceIPAddress_t ipAddr,
+                                  IceAgent_t * pIceAgent,
+                                  IceCandidate_t ** ppCandidate );
+
+IceResult_t Ice_AddSrflxCandidate( const IceIPAddress_t ipAddr,
+                                   IceAgent_t * pIceAgent,
```

Comment:
I'd suggest to put ice agent as the first argument to align other APIs.

Reply:
updated.